src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,166 @@
+package parser;
+
+import exceptions.BocchiException;
+import exceptions.EmptyTaskException;
+import exceptions.InvalidInputException;
+import exceptions.InvalidSyntaxException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+import ui.Ui;
+
+/**
+ * Class used to parse all inputs from the user
+ */
+public class Parser {
+    private static final int EXPECTED_TOKENS_IN_EVENT = 3;
+    private static final String LINE_BREAK = "___________________________________________________";
+    private final Ui ui;
+
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Adds the task to the current task list.
+     *
+     * @param input Remaining input string excluding action
+     * @param taskList Current list of tasks
+     * @return Updated task list
+     */
+    private TaskList addTask(String input, String action, TaskList taskList)
+            throws InvalidSyntaxException {
+        Task task;
+        switch (action) {
+        case "deadline":
+            if (!input.contains("/by")) {
+                throw new InvalidSyntaxException("deadline");
+            }
+            // Further tokenize into action and deadline
+            String[] deadlineTokens = input.split("\\s*/by\\s*");
+            task = new Deadline(deadlineTokens[0], deadlineTokens[1]);
+            break;
+        case "event":
+            if (!input.contains("/from") || !input.contains("/to")) {
+                throw new InvalidSyntaxException("event");
+            }
+            // Further tokenize into action, start time and end time
+            // Regex identified by /to OR /from
+            String[] eventTokens = input.split(
+                    "\\s*/to\\s*|\\s*/from\\s*",
+                    EXPECTED_TOKENS_IN_EVENT
+            );
+            task = new Event(eventTokens[0], eventTokens[1], eventTokens[2]);
+            break;
+        // Task.Todo is the default case
+        default:
+            task = new Todo(input);
```

Comment:
Default case should contain a `break;` statement as well, according to the coding standard.